config.go

```diff
@@ -73,6 +76,13 @@ func SetUnits(uom string) {
 	}
 }
 
+// SetClient sets the http.Client used for requests, returning the previous client.
+func SetClient(client *http.Client) (result *http.Client) {
```

Comment:
I don't see a good reason for this setter to mutate _and return_ the previous client. Ditch the return value here and move your guard clause from `http.go` to here.

Something like:
```
func SetClient(client *http.Client) {
        // allow client to be reset to the default implementation
	if client == nil {
		config.Client = http.DefaultClient
		return
	}
        config.Client = client
}
```

This will keep it so that the client is only mutated from one place so should be easier to add a mutex later in life.

Requires a little more code but testing changes to the config.Client field should be done via `noaa.GetConfig().Client`. I think it's worth the clarity.

Reply:
This comes from a coding habit of mine from c++ days (when you might want to explicitly delete the object or reuse this object elsewhere for some reason).  I can certainly remove this.